commit 8929a57c7497e2786c976277d2ba562ceb7d571d (refs/changes/20/320/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-08-18T20:45:33-07:00 (3 years, 2 months ago)
    
    joyent/node-smartdc-auth#9 requestSigner() regression: duplicated prefixes on keyId

diff --git a/lib/index.js b/lib/index.js
index 5d36102..a9bedb3 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -142,7 +142,13 @@ function requestSigner(options) {
 						user += '/users/' + subuser;
 				}
 				var keyId = '/' + user + '/keys/' + res.keyId;
-				res.keyId = keyId;
+				res = {
+				    algorithm: res.algorithm,
+				    keyId: keyId,
+				    signature: res.signature,
+				    user: user,
+				    subuser: subuser
+				};
 			}
 			cb(err, res);
 		});
